whelktool/scripts/analysis/validator-statistics-auth.groovy

```diff
@@ -0,0 +1,46 @@
+import whelk.JsonLdValidator
+import whelk.JsonLdValidator.Error as JsonError
+import whelk.util.Statistics
+
+class Script {
+    static PrintWriter report
+    static Statistics s = new Statistics(6).printOnShutdown()
+    static JsonLdValidator v
+    static boolean isInitialized = false
+}
+
+Script.report = getReportWriter("report.txt")
+
+selectByCollection('auth') { auth ->
+    synchronized (this) {
```

Comment:
What is "this" in this context (since it's called from a function and not a method)? (Not trying to be snarky, genuine question!)

Is it some global object (which is fine), or is it some temporary object, owned by the executing thread (which makes the synchronized bit not have any effect)?

If the answer is "don't know", consider perhaps synchronizing over some "new Object();" in the global scope instead.

Reply:
If I remember correctly that construction is stolen (by me) from some @olovy - script. It's used to synchronize a block of code rather than a whole method to allow for concurrent execution in the rest of the code: https://stackoverflow.com/questions/13264726/java-syntax-synchronized-this.

Reply:
In this case `this` is the magic groovy script object. (I hope!)

Reply:
@lrosenstrom ? I just did a superlazy search replace bib with auth in the script 😄